packages/cli/src/frontmatter.ts

```diff
@@ -6,14 +6,37 @@
 import * as yaml from 'js-yaml';
 
 const frontMatterRegex = /^-{3}\s*[\n\r](.*?)[\n\r]-{3}\s*[\n\r]+/s;
+const urlIDRegex = /(?<baseURL>.*)\/d\/(?<documentID>[\w-]+)/;
+
+type UrlID = `${string}/d/${string}`;
+
+export function getDocumentID(urlID: UrlID, expectedbaseURL: string) {
+  const match = urlID.match(urlIDRegex);
+  if (match === null) {
+    throw new Error(
+      `Invalid document ID: ${urlID}. Please report this issue to the @mermaidchart/cli maintainers.`,
+    );
+  }
+  // we know that this regex will always return groups on a match
+  const { baseURL, documentID } = match.groups as { baseURL: string; documentID: string };
+  if (baseURL !== expectedbaseURL) {
+    throw new Error(
+      `Your @mermaidchart/cli is configured to use ${expectedbaseURL}, but your diagram is using ${baseURL}`,
+    );
+  }
+  return documentID;
```

Comment:
What if they have a private and public instance, and they're linking a diagram manually?
Might be better to print a warning than throw an error.

In future release, we might have to go the git route and support multiple origins.

Reply:
IMO, you mean printing a warning, then ignoring the diagram?

Hmmmm, I still think we should return a non-zero exit code in this case. I'm guessing if somebody has two different instances, they'd currently have to do something like:

```bash
npx @mermaidchart/cli xxxx ./public/**/*.md
MERMAID_CHART_AUTH_TOKEN='private-token' MERMAID_CHART_BASE_URL='https://private-domain.example' npx @mermaidchart/cli xxxx ./private/**/*.md
```

We could print out a different exit code for this case, but it seems like a lot of work, when most people will probably just a single domain for all their uploads, e.g. `https://mermaidchart.com`!

Reply:
What I had in mind was, we allow users to login to multiple domains (with OAuth), and save the sites and tokens in our config file. Then when we come across those, use the corresponding tokens and proceed.

So the flow would be

```bash
# OneTime
npx @mermaidchart/cli login  # Browser opens, user logs in, we save to config file.
npx @mermaidchart/cli login https://private-domain.example # Browser opens, user logs in, we append to config file.

# Regular use
npx @mermaidchart/cli <other commands> # Uses corresponding auth for each diagram automatically based on the URL.
```

Reply:
Purely a "nice-to-have" feature :)

Reply:
:+1: Definitely!

NPM has this feature, so we should also have it!

Although, it would mean I need to think about the config file format a bit more. Maybe there's an open-source CLI tool we can take inspiration from, like NPM/Pip